hcmanager/classification_filter.go

```diff
@@ -0,0 +1,264 @@
+			return false
+		}
+	} else if expectedCPU.MinimumSpeedMHz > 0 {
+		MinSpeed := bmh.ClockSpeed(float64(expectedCPU.MinimumSpeedMHz / 1000))
+		mgr.Log.Info("", "Provided Minimum ClockSpeed for CPU", MinSpeed, " and fetched ClockSpeed ", cpu.ClockMegahertz)
+		if MinSpeed > cpu.ClockMegahertz {
+			mgr.Log.Info("CPU MIN ClockSpeed did not match")
+			return false
+		}
+
+	}
+
+	return true
+
+}
+
+//checkNICS this function checks the nics details for both min and max parameters
+func checkNICS(mgr HardwareClassificationManager, nics int, expectedNIC *hwcc.Nic) bool {
+
+	if (expectedNIC.MaximumCount > 0) && (expectedNIC.MinimumCount > 0) {
+
+		mgr.Log.Info("", "Provided Minimum Count for NICS", expectedNIC.MinimumCount, " and fetched count ", nics)
+		mgr.Log.Info("", "Provided Maximum count for NICS", expectedNIC.MaximumCount, " and fetched count ", nics)
+		if (expectedNIC.MinimumCount > nics) || (expectedNIC.MaximumCount < nics) {
+
+			mgr.Log.Info("NICS MINMAX count did not match")
+			return false
+		}
+	} else if expectedNIC.MaximumCount > 0 {
+
+		mgr.Log.Info("", "Provided Maximum count for NICS", expectedNIC.MaximumCount, " and fetched count ", nics)
+		if expectedNIC.MaximumCount < nics {
+
+			mgr.Log.Info("NICS MAX count did not match")
+			return false
+		}
+
+	} else if expectedNIC.MinimumCount > 0 {
+
+		mgr.Log.Info("", "Provided Minimum Count for NICS", expectedNIC.MinimumCount, " and fetched count ", nics)
+		if expectedNIC.MinimumCount > nics {
+
+			mgr.Log.Info("NICS MIN count did not match")
+			return false
+		}
+
+	}
+	return true
+}
+
+//checkRAM this function checks the ram details for both min and max parameters
+func checkRAM(mgr HardwareClassificationManager, ram int64, expectedRAM *hwcc.Ram) bool {
+	if (expectedRAM.MaximumSizeGB > 0) && (expectedRAM.MinimumSizeGB > 0) {
+
+		mgr.Log.Info("", "Provided Minimum Size for RAM", expectedRAM.MinimumSizeGB, " and fetched SIZE ", ram)
+		mgr.Log.Info("", "Provided Maximum Size for RAM", expectedRAM.MaximumSizeGB, " and fetched SIZE ", ram)
+		if (expectedRAM.MinimumSizeGB > ram) || (expectedRAM.MaximumSizeGB < ram) {
+			mgr.Log.Info("RAM MINMAX SIZE did not match")
+			return false
+		}
+	} else if expectedRAM.MaximumSizeGB > 0 {
+		mgr.Log.Info("", "Provided Maximum Size for RAM", expectedRAM.MaximumSizeGB, " and fetched SIZE ", ram)
+		if expectedRAM.MaximumSizeGB < ram {
+			mgr.Log.Info("RAM MAX SIZE did not match")
+			return false
+		}
+
+	} else if expectedRAM.MinimumSizeGB > 0 {
+
+		mgr.Log.Info("", "Provided Minimum Size for RAM", expectedRAM.MinimumSizeGB, " and fetched SIZE ", ram)
+		if expectedRAM.MinimumSizeGB > ram {
+
+			mgr.Log.Info("RAM MIN SIZE did not match")
+			return false
+		}
+
+	}
+	return true
+}
+
+//checkDiskDetails this function checks the Disk details for both min and max parameters
+func checkDiskDetails(mgr HardwareClassificationManager, disk []bmh.Storage, expectedDisk *hwcc.Disk) bool {
```

Comment:
disk []bmh.Storage should be defined as disks []bmh.Storage

Reply:
Done